rand/rand.go

```diff
@@ -126,3 +162,23 @@ func StringWithOption(length, option int, prefix, suffix string) string {
 	}
 	return *(*string)(unsafe.Pointer(&buf))
 }
+
+// CheckOption checks if the given option is valid.
+func CheckOption(option, length, numbers int) error {
+	seedLength, ok := optionMappingLen[option]
+	if !ok {
+		return errors.New(errors.Unprocessable, fmt.Sprintf("invalid option: %d", option))
+	}
+
+	min := int(math.Ceil(math.Log(float64(numbers+1)) / math.Log(float64(seedLength))))
```

Comment:
Can you explain a little bit about how do we come to this calculation?

Reply:
@viettd  updated in the comments